app/controllers/stories_controller.rb

```diff
@@ -434,31 +432,40 @@ def set_organization_json_ld
-    @user_json_ld[:worksFor][0][:name] = @user.employer_name if @user.employer_name.present?
-    @user_json_ld[:worksFor][0][:url] = @user.employer_url if @user.employer_url.present?
-    @user_json_ld[:alumniOf] = @user.education if @user.education.present?
-    @user_json_ld[:email] = @user.email if @user.email_public
-    @user_json_ld[:jobTitle] = @user.employment_title if @user.employment_title.present?
-    @user_json_ld[:sameAs].append("https://twitter.com/#{@user.twitter_username}") if @user.twitter_username.present?
-    @user_json_ld[:sameAs].append("https://github.com/#{@user.github_username}") if @user.github_username.present?
-  end
-
-  def set_user_same_as_json_ld
-    @user_json_ld[:sameAs].append(@user.mastodon_url) if @user.mastodon_url.present?
-    @user_json_ld[:sameAs].append(@user.facebook_url) if @user.facebook_url.present?
-    @user_json_ld[:sameAs].append(@user.youtube_url) if @user.youtube_url.present?
-    @user_json_ld[:sameAs].append(@user.linkedin_url) if @user.linkedin_url.present?
-    @user_json_ld[:sameAs].append(@user.behance_url) if @user.behance_url.present?
-    @user_json_ld[:sameAs].append(@user.stackoverflow_url) if @user.stackoverflow_url.present?
-    @user_json_ld[:sameAs].append(@user.dribbble_url) if @user.dribbble_url.present?
-    @user_json_ld[:sameAs].append(@user.medium_url) if @user.medium_url.present?
-    @user_json_ld[:sameAs].append(@user.gitlab_url) if @user.gitlab_url.present?
-    @user_json_ld[:sameAs].append(@user.instagram_url) if @user.instagram_url.present?
-    @user_json_ld[:sameAs].append(@user.twitch_username) if @user.twitch_username.present?
-    @user_json_ld[:sameAs].append(@user.website_url) if @user.website_url.present?
+  def user_works_for
+    # For further examples of the worksFor and disambiguatingDescription properties,
+    # please refer to this link: https://jsonld.com/person/
+    return unless @user.employer_name.presence || @user.employer_url.presence
+
+    {
+      "@type": "Organization",
+      "name": @user.employer_name,
+      "url": @user.employer_url
+    }.reject { |_, v| v.blank? }
+  end
+
+  def user_disambiguating_description
+    [@user.mostly_work_with, @user.currently_hacking_on, @user.currently_learning].compact
+  end
+
+  def user_same_as
+    # For further information on the sameAs property, please refer to this link:
+    # https://schema.org/sameAs
+    [
+      @user.twitter_username.presence ? "https://twitter.com/#{@user.twitter_username}" : nil,
+      @user.github_username.presence ? "https://github.com/#{@user.github_username}" : nil,
+      @user.mastodon_url,
+      @user.facebook_url,
+      @user.youtube_url,
+      @user.linkedin_url,
+      @user.behance_url,
+      @user.stackoverflow_url,
+      @user.dribbble_url,
+      @user.medium_url,
+      @user.gitlab_url,
+      @user.instagram_url,
+      @user.twitch_username,
+      @user.website_url,
+    ].compact
```

Comment:
keep in mind that compact only removes `nil`, not empty string. Don't know which of these can be empty but it could happen that you have `["value", "", "value"]` in the array. I think :D

Reply:
`your_array.reject { |item| item.blank? }` might help here ;)

Reply:
That's a good point! Maybe I should replace this with `.reject` instead... 🤔

Reply:
I think so @vaidehijoshi !